precompiles/parachain-staking/src/lib.rs

```diff
@@ -196,7 +196,84 @@ where
 		Ok(selected_candidates)
 	}
 
+	#[precompile::public("delegationAmount(address,address)")]
+	#[precompile::view]
+	fn delegation_amount(
+		handle: &mut impl PrecompileHandle,
+		delegator: Address,
+		candidate: Address,
+	) -> EvmResult<U256> {
+		// Fetch info.
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let (candidate, delegator) = (
+			Runtime::AddressMapping::into_account_id(candidate.0),
+			Runtime::AddressMapping::into_account_id(delegator.0),
+		);
+		let amount: U256 = if let Some(state) =
+			pallet_parachain_staking::Pallet::<Runtime>::delegator_state(&delegator)
+		{
+			// get the delegation amount
+			if let Some(pallet_parachain_staking::Bond { amount, .. }) = state
+				.delegations
+				.0
+				.into_iter()
+				.find(|b| b.owner == candidate)
+			{
+				amount.into()
+			} else {
+				log::trace!(
+					target: "staking-precompile",
+					"Delegation for {:?} not found, so delegation amount is 0",
+					candidate
+				);
+				U256::zero()
+			}
+		} else {
+			log::trace!(
+				target: "staking-precompile",
+				"Delegator state for {:?} not found, so delegation amount is 0",
+				delegator
+			);
+			U256::zero()
+		};
```

Comment:
I'd suggest to drop the trace logs since the function is user-callable and as such can be called by virtually any input. Returning `U256::zero()` is a good behavior to handle arbitrary input.

```suggestion
		let amount = pallet_parachain_staking::Pallet::<Runtime>::delegator_state(&delegator)
			.and_then(|state| {
				state
					.delegations
					.0
					.into_iter()
					.find(|b| b.owner == candidate)
			})
			.map_or(
				U256::zero(),
				|pallet_parachain_staking::Bond { amount, .. }| amount.into(),
			);
```